### DISCUSSION TITLE
incremental linking with watcom? busybox-w32 compilation by Open Watcom

### DISCUSSION THREAD

**Discussion by staalmannen** [open-watcom/open-watcom-v2#127] [Show and tell]
Hi all.

For fun, I have tried to build busybox-w32 with Open Watcom
https://github.com/staalmannen/busybox-w32/commits/watcom

(if you are curious you are welcome to play with it - I am sure that people with more insight could un-do some stupid hacks I have made. Do not let the ammount of changes fool you - most changes have been reverted in later commits so the size of the delta will be reduced significantly when I do a rebase)

At this first stage, it still depends on GCC for HOSTCC, a Linux host and GNU make. So basically using OpenWatcom as a cross compiler instead of MinGW. Later stages could aim for native build under Windows and reduction of external dependencies (gnu make, unix-like tools).

The status right now is that I manage tobuild everything up until the final linking. The issue I have is that kbuild generates "built-in.o" files that need to be generated by  incremental linking (ld -r). When I google for it it seems like this should also exist for wlink but I have not figured out how to enable it. "incremental" is not among the documented options. for $LD, I use owcc, so I have tried passing options to the linker using "-Wl,"

**Comment by jmalak** [comment]
OW linker incremental linking is internal linker process to spead up linking process if only a few objects are changed. It is done for final executable but not for *.o files (generated by compiler). Anyway incremental linking has sense for development phase, release versio should be created by full link.

**Comment by staalmannen** [comment]
I managed to work around the incremental linking of .o files by using library archives instead.
From today on, busybox for win32 can be built with Open Watcom :+1:

https://github.com/staalmannen/busybox-w32/tree/watcom

**Comment by jmalak** [comment]
Great. Thanks for info.
